### DISCUSSION TITLE
LLaVA with GGUF files (and --mmproj argument)?

### DISCUSSION THREAD

**Discussion by benmayersohn** [vllm-project/vllm#8341] [Q&A]
Hi everyone! I'm coming from `llama.cpp` which currently doesn't support serving multi-modal models (it was [removed](https://github.com/ggerganov/llama.cpp/issues/8595#issuecomment-2241638447) and hopefully it will return soon).

I'm used to the GGUF format, which is only recently supported by vLLM. In `llama.cpp` you would do something like this:

```bash
./llama-server -t ${N_THREADS} -m ${LLAVA_MODEL} -c ${N_CTX} --host ${HOST} \
  --port ${PORT} -ngl ${N_GPU_LAYERS} --mmproj ${MMPROJ} --temp 0.1 \
  -p $'### User: What do you see?\n### Assistant: '
``` 

where

- `${MMPROJ}` is a multi-modal projector that aligns the vision and text data (per [this overview](https://github.com/neobundy/Deep-Dive-Into-AI-With-MLX-PyTorch/blob/master/deep-dives/006-llava/README.md))
-  `${LLAVA_MODEL}` is the Llava model itself

and both are two distinct GGUF files. Does anyone know what the equivalent command/arguments be when serving via `python -m vllm.entrypoints.openai.api_server`?

Thanks!

**Comment by JadTawil77** [comment]
Any news on this?

I am attempting to use the gemma 4b multimodal GGUF model:

gemma-3-4b-it-q4_0.gguf"  # Main model file
mmproj-model-f16-4B.gguf"  # Multimodal projection file
